dask_expr/_collection.py

```diff
@@ -662,10 +662,7 @@ def shift(self, periods=1, freq=None, axis=0):
             raise TypeError("periods must be an integer")
 
         axis = _validate_axis(axis)
-        if axis == 1:
-            raise NotImplementedError("shift on axis 1 not supported yet")
-
-        return new_collection(Shift(self.expr, periods, freq))
+        return new_collection(Shift(self.expr, periods, freq, axis))
```

Comment:
No, please call map_partitions on this level, don't even go into the Shift expression for axis=1

Reply:
Ah, I see now. https://github.com/dask-contrib/dask-expr/pull/473/commits/5015f8f23d1698b03ebb40c3c3056ec533e627b3